Firestore/Example/Tests/Integration/API/FIRDatabaseTests.mm

```diff
@@ -1402,4 +1402,56 @@ - (void)testCanRemoveListenerAfterShutdown {
   [listenerRegistration remove];
 }
 
+- (void)testWaitForPendingWritesCompletes {
+  FIRDocumentReference *doc = [self documentRef];
+  FIRFirestore *firestore = doc.firestore;
+
+  [self disableNetwork];
+
+  [doc setData:@{@"foo" : @"bar"}];
+  [firestore waitForPendingWritesWithCompletion:
+                 [self completionForExpectationWithName:@"Wait for pending writes"]];
+
+  [firestore enableNetworkWithCompletion:[self completionForExpectationWithName:@"Enable network"]];
+  [self awaitExpectations];
+}
+
+- (void)testWaitForPendingWritesFailsWhenUserChanges {
+  FIRApp *app = testutil::AppForUnitTesting(util::MakeString([FSTIntegrationTestCase projectID]));
```

Comment:
Is there any reason why this needs to be a manually constructed `firestore` and `app` combo? It seems like you could just use `self.db`?

Reply:
Done.

This is a result of a temporary Mock/FakeCredentialsProvider implementation. I updated the implementation but forgot to update the test.